### PR TITLE
Recommend a default size of 1 for the `SimpleFixedSizeExemplarReservoir`

### PR DESCRIPTION
**Pull request by MrAlias**
This reservoir type is used for all aggregations other than a histogram with more than one bucket. Each attribute set the aggregation records will have an instance of this reservoir. Therefore, limiting this to a small value by default when enabled is preferable.

This does not address the way a user will configure this value. That is left for a future PR/Issue.

Part of #2421

### Additional context

For similar aggregation types, [OpenMetrics](https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#counter) allows one exemplar.

It looks like [OpenCensus](https://github.com/census-instrumentation/opencensus-specs/blob/405abc2a2fd5a05bb10a8c7b766456726c30ee05/stats/Exemplars.md#retention-in-views) expected libraries to "retain and export at least [one] exemplar". But it is a little unclear if this is comparable.

### CODE REVIEW

**Review by yurishkuro — APPROVED** [review]

**Review by jack-berg — APPROVED** [review]

File: spec-compliance-matrix.md [hunk 1]
```diff
@@ -208,6 +208,7 @@ formats is required. Implementing more than one format is optional.
 | The metrics SDK provides an `ExemplarReservoir` interface or extension point.                                                                                          | X        |    | -    |     | -      |      |        | +   |      |     | -    |       |
 | An `ExemplarReservoir` has an `offer` method with access to the measurement value, attributes, `Context` and timestamp.                                                | X        |    | -    |     | -      |      |        | +   |      |     | -    |       |
 | The metrics SDK provides a `SimpleFixedSizeExemplarReservoir` that is used by default for all aggregations except `ExplicitBucketHistogram`.                           |          |    | +    |     | -      |      |        | +   |      |     | -    |       |
+| The default size of the `SimpleFixedSizeExemplarReservoir` is `1`.                                                                                                     |          |    |      |     |        |      |        |     |      |     |      |       |
```
Comment (cijothomas) [inline spec-compliance-matrix.md:211]: @utpilla 
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Metrics/MetricPoint.cs#L29-L30 Lets update OTel .NET to match this, once this PR gets merged.

**Review by cijothomas — APPROVED** [review]

**Review by jsuereth — APPROVED** [review]
I'm not fully convinced the limit of "1" by OpenMetrics is a great default.  IIUC - It's more a limitation of their protocol (only one exemplar per "point").

Specifically -

SimpleFixedSizeExemplarReservoir is *likely* the right reservoir for ExponentialHistograms.   For these, we should definitely encourage a larger pool size.

Limiting to 1 is fine for now, but with ExponentialHistograms moving towards stable I'd encourage us to call out a different default for those.

**Comment by MrAlias** [comment]
> SimpleFixedSizeExemplarReservoir is _likely_ the right reservoir for ExponentialHistograms. For these, we should definitely encourage a larger pool size.

Would it make sense in this case to use a default size equal to the exponential histograms max size configuration?

I can update for that aggregator in a follow on up PR if that makes sense @jsuereth.

**Comment by jsuereth** [comment]
> Would it make sense in this case to use a default size equal to the exponential histogram's max size configuration?

I actually think the exponential histogram's default may be a bit too large right now, at least if we also track exemplars there.  I'd suggest limiting to a range of 5-20 exemplars (or some division of the exponential histogram's max size) purely from an overhead standpoint on tracking these points.  I don't have hard numbers to back this up, just some anecdotes.

Here's some VERY lazy math with naive assumptions:

### Explicit Bucket histograms
- Default 15 boundaries, 16 buckets
- 1:1 exemplar to bucket
- Likely storage per bucket count = 8 bytes
- Max, min, sum storage = 8*3 = 24 bytes
- Memory overhead without Exemplars: 16*8 + 8*3 = 152 bytes
- Likely storage per exemplar (best-case pointer to shared attributes, pointer to shared context + measurement + timestamp) = 32 bytes
- Memory head just for Exemplars = 512 bytes
- Total overhead / histogram = 664 bytes

### Exponential Bucket Histogram
- Default 320 buckets + overflow (Let's assume negative measurements don't exist)
- Likely storage per bucket count = ~2 bytes (assuming we're using "adapting integer" algorithm where we scale arrays as counts grow, and we're in a sweet spot of measurement-counts)
- max, min, sum storage = 8*3 bytes
- Memory overhead without exemplars - ~640 + 24 = 664 bytes

Given the key size differential, I don't think we should be spending quite as much memory overhead on exemplars.  Particularly because Exponential Buckets are intended to give us very high accuracy percentiles.  I think we're better of attempting to grab lower number but higher entropy exemplars.

> I can update for that aggregator in a follow up PR if that makes sense @jsuereth.

that's good with me!

**Comment by MrAlias** [comment]
> I can update for that aggregator in a follow up PR [...]

Tracking with https://github.com/open-telemetry/opentelemetry-specification/issues/3674

**Pull request merged**